controllers/secrets-webhook/secretswebhook.go

```diff
@@ -0,0 +1,162 @@
+package secrets_webhook
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"net/http"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	log "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-pod-secrets-inject,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io,admissionReviewVersions=v1,sideEffects=None
+
+func SetupSecretsInjectorWebhook(mgr ctrl.Manager) {
```

Comment:
I think you'll need to call this as part of the envtest setup around [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/test/testenv.go#L104), I think that's what is causing the bulk of the test failures in the envtests right now.

Reply:
Looks like you may need to do the same in the Medusa controller tests too FYI.